src/metrics/mod.rs

```diff
@@ -74,13 +152,56 @@ impl Metrics {
         }
     }
 
-    pub fn delta(&self, snapshot: &mut Snapshot) -> u64 {
-        let curr = self.metric().value();
-        let prev = snapshot.prev.insert(*self, curr).unwrap_or(0);
+    pub fn delta(&self, snapshot: &mut MetricsSnapshot) -> u64 {
+        let curr = self.counter().value();
+        let prev = snapshot.counters.insert(*self, curr).unwrap_or(0);
         curr.wrapping_sub(prev)
     }
 }
 
+#[allow(clippy::enum_variant_names)]
+#[derive(Eq, Hash, PartialEq, Copy, Clone)]
+pub enum HistogramMetric {
+    PubsubLatency,
+    PubsubPublishLatency,
+    ResponseLatency,
+}
+
+impl HistogramMetric {
+    pub fn histogram(&self) -> &metriken::AtomicHistogram {
+        match self {
+            Self::PubsubLatency => &PUBSUB_LATENCY,
+            Self::PubsubPublishLatency => &PUBSUB_PUBLISH_LATENCY,
+            Self::ResponseLatency => &RESPONSE_LATENCY,
+        }
+    }
+
+    pub fn delta(&self, snapshot: &mut MetricsSnapshot) -> Option<histogram::Snapshot> {
```

Comment:
Is this delta function actually meaningfully used anywhere? The only callers I can find are from output/log and it initializes a new HistogramMetric on every invocation.